Tests create ILM history index while template is missing

**Build scan**: https://gradle-enterprise.elastic.co/s/sd4va4diu23ps

**Repro line**: `REPRODUCE WITH: ./gradlew ':x-pack:plugin:watcher:qa:rest:yamlRestTest' --tests "org.elasticsearch.smoketest.WatcherYamlRestIT.test {p0=painless/40_exception/Test awesome painless exceptions are returned including the script_stack field}" -Dtests.seed=41568401864EE36D -Dtests.security.manager=true -Dtests.locale=fr-CH -Dtests.timezone=Africa/Monrovia -Druntime.java=11`

**Reproduces locally?**: No

**Applicable branches**: Observed on `master`

**Failure history**: This particular test has failed a couple of times recently, but the problem is more general.

**Failure excerpt**:

```
java.lang.AssertionError: expected [2xx] status code but api [cluster.health] returned [408 Request Timeout] [{"cluster_name":"yamlRestTest","status":"yellow","timed_out":true,"number_of_nodes":1,"number_of_data_nodes":1,"active_primary_shards":1,"active_shards":1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1,"delayed_unassigned_shards":0,"number_of_pending_tasks":0,"number_of_in_flight_fetch":0,"task_max_waiting_in_queue_millis":0,"active_shards_percent_as_number":50.0}] |  
  | at org.junit.Assert.fail(Assert.java:88) |  
  | at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:296) |  
  | at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:407)
```

However the actual problem can be seen in the cluster logs:

```
[2020-11-12T11:31:04,880][INFO ][o.e.c.m.MetadataIndexTemplateService] [yamlRestTest-0] removing index template [ilm-history]
[2020-11-12T11:31:04,907][INFO ][o.e.c.m.MetadataCreateIndexService] [yamlRestTest-0] [ilm-history-5] creating index, cause [auto(bulk api)], templates [], shards [1]/[1]
[2020-11-12T11:31:04,944][INFO ][o.e.c.m.MetadataIndexTemplateService] [yamlRestTest-0] adding index template [ilm-history] for index patterns [ilm-history-5*]
```

The test cluster only has one node, but the newly-created `ilm-history-5` index has a replica, so we can never achieve `green` health. The `ilm-history` index template normally ensures this index doesn't have replicas but unfortunately it's missing at the time the index is created.

--- a/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java
@@ -1315,6 +1315,7 @@ public abstract class ESRestTestCase extends ESTestCase {
             case "synthetics-mappings":
             case ".snapshot-blob-cache":
             case ".deprecation-indexing-template":
+            case "ilm-history":
                 return true;
             default:
                 return false;

